new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Voyager.java

```diff
@@ -0,0 +1,155 @@
+
+package oracle.weblogic.kubernetes.actions.impl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.kubernetes.client.custom.IntOrString;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1HTTPIngressPath;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1HTTPIngressRuleValue;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1Ingress;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressBackend;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressList;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressRule;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressSpec;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import oracle.weblogic.kubernetes.actions.impl.primitive.Helm;
+import oracle.weblogic.kubernetes.actions.impl.primitive.HelmParams;
+import oracle.weblogic.kubernetes.actions.impl.primitive.Kubernetes;
+
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+
+/**
+ * Utility class for Voyager ingress controller.
+ */
+public class Voyager {
+  /**
+   * Install Voyager Helm chart.
+   *
+   * @param params the parameters to Helm install command such as release name, namespace, repo url or chart dir,
+   *               chart name and chart values
+   * @return true on success, false otherwise
+   */
+  public static boolean install(VoyagerParams params) {
+    return Helm.install(params.getHelmParams(), params.getValues());
+  }
+
+  /**
+   * Upgrade Voyager Helm release.
+   *
+   * @param params the parameters to Helm upgrade command such as release name, namespace and chart values to override
+   * @return true on success, false otherwise
+   */
+  public static boolean upgrade(VoyagerParams params) {
+    return Helm.upgrade(params.getHelmParams(), params.getValues());
+  }
+
+  /**
+   * Uninstall Voyager Helm release.
+   *
+   * @param params the parameters to Helm uninstall command such as release name and namespace
+   * @return true on success, false otherwise
+   */
+  public static boolean uninstall(HelmParams params) {
+    return Helm.uninstall(params);
+  }
+
+  /**
+   * Create an ingress for the WebLogic domain with domainUid in the specified domain namespace.
+   * The ingress host is set to 'domainUid.clusterName.test'.
+   *
+   * @param ingressName name of the ingress to be created
+   * @param domainNamespace the WebLogic domain namespace in which the ingress will be created
+   * @param domainUid the WebLogic domainUid which is backend to the ingress
+   * @param clusterNameMsPortMap the map with key as cluster name and value as managed server port of the cluster
+   * @return list of ingress hosts or null if got ApiException when calling Kubernetes client API to create ingress
+   */
+  public static List<String> createIngress(String ingressName,
```

Comment:
I added more check in Sticky session test suite. Let me move it over here

Reply:
Seems that we add the checks in CommonTestUtils.java. There are several checks there to make sure the ingress is up and ready to use. If I move all these checks to Voyager.java(now Ingress.java), it would make the class to looks to complex. Can you please check CommonTestUtils.installVoyagerIngressAndVerify for those checks, see if you like them?